v1/writable/image_test.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writable_test
```

Comment:
Is there a reason this shouldn't be in `package writable`?

Reply:
Forces you to test only exported types & methods. Thus you get a compiler error if there’s a leak. 

The motivations for testing like this enable refactoring internals cleanly.

Reply:
I'll change it to be consistent with the project